java-di-bench/app/controllers/BenchController.java

```diff
@@ -50,9 +65,23 @@ public Result downloadChunked(final int length) {
     return ok().chunked(Source.from(chunkBuffer));
   }
 
-  @BodyParser.Of(BodyParser.Raw.class)
   public Result upload() {
-    return ok("upload"); // TODO: Verify upload happened
+    MultipartFormData<File> body = request().body().asMultipartFormData();
+    FilePart<File> uploadedFile = body.getFiles().get(0);
+
+    if (uploadedFile == null) {
+      return badRequest("No file");
+    }
+
+    String filename = uploadedFile.getFilename();
+    File source = uploadedFile.getFile();
+    // Do nothing with the file.
+    return ok("It works!");
+  }
+
+  @BodyParser.Of(BodyParser.Raw.class)
+  public Result uploadRaw() {
+    return ok("It works"); // TODO: Verify upload happened
```

Comment:
Actually, I don't think we need anything here, just like we do in the Scala benchs.